Babylon/groups/api/groups/workspace/commands/delete.py

```diff
@@ -0,0 +1,122 @@
+
+
+@command()
+@allow_dry_run
+@pass_workspace_api
+@timing_decorator
+@require_deployment_key("organization_id", "organization_id")
+@option(
+    "-f",
+    "--force",
+    "force_validation",
+    is_flag=True,
+    help="Don't ask for validation before delete",
+)
+@option(
+    "-e",
+    "--use-working-dir-file",
+    "use_working_dir_file",
+    is_flag=True,
+    help="Should the path be relative to the working directory ?",
+    type=bool,
+)
+@option(
+    "-w",
+    "--workspace-file",
+    "workspace_file",
+    help="In case the workspace id is retrieved from a file",
+)
+@argument("workspace_id", required=False)
+def delete(
+    workspace_api: WorkspaceApi,
+    organization_id: str,
+    workspace_file: Optional[str] = None,
+    workspace_id: Optional[str] = None,
+    dry_run: Optional[bool] = False,
+    use_working_dir_file: Optional[bool] = False,
+    force_validation: Optional[bool] = False,
+):
+    """Unregister a workspace via Cosmotech APi."""
+
+    if dry_run:
+        logger.info("DRY RUN - Would call workspace_api.delete_workspace")
+        return
+
+    if not workspace_id:
+        if not workspace_file:
+            logger.error(
+                "Error: No id passed as argument or option use -d option to pass an json or yaml file containing an workspace id."
+            )
+            return
+
+        converted_workspace_content = get_api_file(
+            api_file_path=workspace_file,
+            use_working_dir_file=use_working_dir_file,
+            logger=logger,
+        )
+        if not converted_workspace_content:
+            logger.error("Error : can not get Workspace definition, please check your file")
+            return
+
+        if converted_workspace_content["id"]:
+            workspace_id = converted_workspace_content["id"]
+        elif converted_workspace_content["workspace_id"]:
+            workspace_id = converted_workspace_content["workspace_id"]
+        else:
+            logger.error(f"Error: Could not found workspace id in {workspace_file}.")
+            return
+
+    try:
+        workspace_api.find_workspace_by_id(workspace_id=workspace_id, organization_id=organization_id)
+    except NotFoundException:
+        logger.error(f"Workspace {workspace_id} does not exists in organization {organization_id}.")
+        return
+    except UnauthorizedException:
+        logger.error("Unauthorized access to the cosmotech api")
+        return
+
+    if not force_validation:
+
+        if not confirm(
+            f"You are trying to delete workspace {workspace_id} workspaces of organization {organization_id} \nDo you want to continue?"
+        ):
+            logger.info("Workspace deletion aborted.")
+            return
+
+        confirm_workspace_id = prompt("Confirm workspace id ")
+        if confirm_workspace_id != workspace_id:
+            logger.error("The workspace id you have type didn't mach with workspace you are trying to delete id")
+            return
```

Comment:
I wonder if we could make this bit of code generic and put it in an utils/interactive function

Reply:
Good idee ! Why not